File: flame/pytest/ui.py
# -*- coding: utf-8 -*-
"""
@date: 2025
@author: Samantha C Pendleton
@description: shiny dashboard
@GitHub: github.com/sap218/jabberwocky

@useful links:
    # https://shiny.posit.co/py/docs/overview.html
"""

# ui.py
import shiny
from shiny import ui

# ui.py
import shiny
from shiny import ui

def ui():
    return ui.page_fluid(
        ui.theme('cosmo'),  # theme like in R Shiny
        ui.panel_title("Jabberwocky"),
        
        ui.layout_sidebar(
            ui.panel_sidebar(
                ui.help_text("Data"),
                ui.input_file("corpus", "Upload corpus", accept=".txt"),
                ui.input_radio_buttons("filter_level", "Stopword filtering", choices=["light", "heavy"]),
                ui.input_radio_buttons("output_format", "Output format", choices=["wtags", "grep", "invertedgrep"]),
            ),
            
            ui.panel_main(
                ui.navbar_page(
                    "Navigation",
                    
                    ui.nav_panel("Corpus", 
                                 ui.layout_sidebar(
                                     ui.panel_sidebar(
                                         ui.help_text("Features"),
                                         ui.input_radio_buttons("cm", "Wordcloud scheme", choices=["Set3", "viridis"]),
                                         ui.input_colour_picker("highlightcolour", "Highlighter", value="#00bcd4")
                                     ),
                                     ui.panel_main(
                                         ui.output_text("fileContent")
                                         #ui.div(style="max-height:200px; overflow-y:scroll; border:0.5px solid #ccc; padding:0px;", 
                                         #       ui.output_text("fileContent"))
                                     )
                                 )
                    )
                )
            )
        )
    )

File: flame/rtest/ui.R
# UI ----------------------------------------------------------------------

library(shinythemes) # theme
library(colourpicker) # cyan

ui <- fluidPage(
  theme = shinytheme("cosmo"), # list of nice themes: cosmo, flatly, paper, yeti
  
  titlePanel("Jabberwocky"),
  
  fluidRow(
    column(3,
           helpText("Data"),
      
           fileInput("corpus", "Upload corpus", accept = ".txt"),
           fileInput("concepts", "Upload words of interest", accept = ".txt"),
           
           # selectInput("filter_level", "Stopword filtering", c("light","heavy")),
           radioButtons("filter_level", "Stopword filtering", c("light","heavy")),
           radioButtons("output_format", "Output format", c("wtags","grep","invertedgrep")),

           #output_name, stats_output_name, plot_output_name, cyannotator_output_name
           
    ), # end of left panel
    
    column(9,

           navbarPage("Navigation",
                      
                      tabPanel("Data",
                               sidebarLayout(
                                 position = "right",
                                 sidebarPanel(
                                   helpText("Features"),
                                   radioButtons("cm", "Wordcloud scheme", c("Set3","viridis")),
                                   colourInput("highlightcolour", "Highlighter", value="#00bcd4")
                                 ), # end of right panel
                                 
                                 mainPanel(
                                   
                                   #textOutput("whatFile")
                                   
                                   h4("Corpus"),
                                   tags$div(
                                     style="max-height:200px; overflow-y:scroll; border:0.5px solid #ccc; padding:0px;",
                                     verbatimTextOutput("corpusContent")
                                   ),
                                   
                                   br(),
                                   
                                   h4("Words of interest"),
                                   tags$div(
                                     style="max-height:200px; overflow-y:scroll; border:0.5px solid #ccc; padding:0px;",
                                     verbatimTextOutput("conceptsContent")
                                   )
                                   
                                 ) # end of main panel
                               ) # end of panel layout
                      )#, # end panel
                      
# Next Section ------------------------------------------------------------

                      # tabPanel("Other panel",
                      # ) # end panel
                      
           ) # end of navbar
    ) # end of centre panel
  ) # end of fluid rows
  
# End ---------------------------------------------------------------------

)